src/plugins/explore/public/application/utils/state_management/actions/processors/trace_aggregation_processor.ts

```diff
@@ -0,0 +1,471 @@
+              }
+            }
+          }
+        }
+      } catch (error) {
+        // Fall back to converting the interval string if histogram config fails
+        intervalMs = convertIntervalToMs(interval);
+      }
+    } else {
+      // No timeFieldName or uiSettings, fall back to converting the interval string
+      intervalMs = convertIntervalToMs(interval);
+    }
+
+    if (requestAggResults) {
+      result.requestChartData = processRequestCountData(
+        requestAggResults,
+        dataset,
+        minTime,
+        maxTime,
+        intervalMs,
+        timeField,
+        xAxisFormat
+      );
+    }
+
+    if (errorAggResults) {
+      result.errorChartData = processErrorCountData(
+        errorAggResults,
+        dataset,
+        minTime,
+        maxTime,
+        intervalMs,
+        timeField,
+        xAxisFormat
+      );
+    }
+
+    if (latencyAggResults) {
+      result.latencyChartData = processLatencyData(
+        latencyAggResults,
+        dataset,
+        minTime,
+        maxTime,
+        intervalMs,
+        timeField,
+        xAxisFormat
+      );
+    }
+
+    result.bucketInterval = bucketInterval;
+  } catch (error) {
+    // Error during processing
+  }
+
+  return result;
+};
+
+function convertIntervalToMs(interval: string): number {
+  // Handle unit-only intervals (m, s, h, d) - transform to 1 unit
+  if (/^[smhd]$/.test(interval)) {
+    const transformedInterval = `1${interval}`;
+    return convertIntervalToMs(transformedInterval);
+  }
+
+  const match = interval.match(/^(\d+)([smhd])$/);
+  if (!match) {
+    return 300000; // Default 5 minutes
+  }
+
+  const [, value, unit] = match;
+  const num = parseInt(value, 10);
+
+  let result: number;
+  switch (unit) {
+    case 's':
+      result = num * 1000;
+      break;
+    case 'm':
+      result = num * 60 * 1000;
+      break;
+    case 'h':
+      result = num * 60 * 60 * 1000;
+      break;
+    case 'd':
+      result = num * 24 * 60 * 60 * 1000;
+      break;
+    default:
+      result = 300000;
+  }
+
+  return result;
+}
+
+export function extractPPLIntervalMs(results: ISearchResult, fallbackMs: number): number {
+  if (!results.hits?.hits?.[0]?._source) {
+    return fallbackMs;
+  }
+
+  const firstHit = results.hits.hits[0];
+  const spanTimeKey = Object.keys(firstHit._source).find((key) => key.startsWith('span('));
+
+  if (!spanTimeKey) {
+    return fallbackMs;
+  }
+
+  const intervalMatch = spanTimeKey.match(/span\([^,]+,\s*(\d+)([smhd])\)/);
+  if (!intervalMatch) {
+    return fallbackMs;
+  }
+
+  const [, value, unit] = intervalMatch;
+  const num = parseInt(value, 10);
+
+  switch (unit) {
+    case 's':
+      return num * 1000;
+    case 'm':
+      return num * 60 * 1000;
+    case 'h':
+      return num * 60 * 60 * 1000;
+    case 'd':
+      return num * 24 * 60 * 60 * 1000;
+    default:
+      return fallbackMs;
+  }
+}
+
+function processRequestCountData(
```

Comment:
The three processing functions (processRequestCountData, processErrorCountData, processLatencyData) in trace_aggregation_processor.ts share a lot of code. 

Does it make sense to refactor them into a single function:

```
Refactor into a single generic function:

  interface ChartProcessConfig {
    metricField: string | string[];
    yAxisLabel: string;
    transformer?: (value: number) => number;
  }

  function processChartData(
    results: ISearchResult,
    config: ChartProcessConfig,
    dataset: DataView,
    minTime: number,
    maxTime: number,
    intervalMs: number,
    timeField: string,
    xAxisFormat: any
  ): ChartData { /* ... */ }
```